cibuildwheel/action.yml

```diff
@@ -0,0 +1,169 @@
+name: cibuildwheel
+description: 'Installs pipx to run cibuildwheel on the current runner with some opinionated settings for RAPIDS'
+inputs:
+  package-dir:
+    description: 'Path to python package'
+    required: true
+  output-dir:
+    description: 'Path to output wheels'
+    required: true
+  python-version:
+    description: 'Python version e.g. 3.8'
+    required: true
+  python-package-cuda-suffix:
+    description: 'Add a -cu11-style suffix to wheel name'
+    required: false
+    default: ''
+  python-package-versioneer-override:
+    description: 'Override versioneer with a clean string'
+    required: false
+    default: ''
+  auditwheel-repair-command-amd64:
+    description: 'Auditwheel repair command amd64'
+    required: true
+  auditwheel-repair-command-arm64:
+    description: 'Auditwheel repair command amd64'
+    required: true
+  internal-pypi-user:
+    description: 'Username for pypi.k8s.rapids.ai'
+    required: true
+  internal-pypi-pass:
+    description: 'Password for pypi.k8s.rapids.ai'
+    required: true
+  manylinux-container-amd64:
+    description: 'Specify which manylinux container to use for amd64'
+    required: true
+  manylinux-container-arm64:
+    description: 'Specify which manylinux container to use for arm64'
+    required: true
+  cibw-before-all-amd64:
+    description: 'Extra CIBW_BEFORE_ALL_LINUX commands for amd64'
+    required: false
+    default: ''
+  cibw-before-all-arm64:
+    description: 'Extra CIBW_BEFORE_ALL_LINUX commands for arm64'
+    required: false
+    default: ''
+  cibw-before-build:
+    description: 'Extra CIBW_BEFORE_BUILD_LINUX commands'
+    required: false
+    default: ''
+  cibw-environment:
+    description: 'CIBW_ENVIRONMENT'
+    required: false
+    default: ''
+  skbuild-configure-options:
+    description: 'SKBUILD_CONFIGURE_OPTIONS'
+    required: false
+    default: ''
+  skbuild-build-options:
+    description: 'SKBUILD_BUILD_OPTIONS'
+    required: false
+    default: ''
+  sccache-aws-access-key-id:
+    required: true
+    type: string
+  sccache-aws-secret-access-key:
+    required: true
+    type: string
+  cumlprims_mg-private-deploy-key:
+    required: false
+    type: string
+    default: ''
+  cugraph-ops-private-deploy-key:
+    required: false
+    type: string
+    default: ''
+
+runs:
+  using: composite
+  steps:
+    - name: Install python3.8, venv, pip on runner container
+      run: sudo apt update -y && sudo apt install -y python3.8-venv python3-venv python3-pip
+      shell: bash
+
+    - name: Install pipx
+      run: |
+        python3 -m pip install pipx
```

Comment:
Note again: thin launcher container can have cibuildwheel baked in, avoid using pipx unnecessarily